### DISCUSSION TITLE
Disable ilia pages that are not supported by the window manager in favour of cross wm usage

### DISCUSSION THREAD

**Discussion by SoumyaRanjanPatnaik** [regolith-linux/regolith-desktop#765] [Ideas]
I think ilia is an excellent choice as an application launcher. But pages like the keybindings page and the windows page won't exactly work with other window managers (eg. bspwm, dwm, etc). So I think it would make sense to disable these pages depending on the window manager to avoid showing blank pages to the users. We can instead print an error on the console to tell the user that these features are not supported by their window manager.

**Comment by kgilmer** [comment]
It's a good suggestion.  `ilia` ideally can be used freely outside of Regolith.  There are two other approaches to this problem we may want to consider: Implement a plugin architecture and have each page be a plugin.  Plugins may be desktop specfic, etc.  The other approach would be to change the build such that each page was extracted as a separate executable, or the build system could control which pages get compiled into the app.  This may allow us to create different "distributions" of ilia suited to specific use cases.  

The plugin architecture approach is more flexibly but may have a cost in terms of startup performance.  As an app launcher, `ilia`'s only useful if it's fast.  The build architecture seems relatively simple in comparison but is obviously less flexible.
